src/Simplify.Web/Model/Validation/Attributes/MaxAttribute.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Reflection;
+using Simplify.DI;
+
+namespace Simplify.Web.Model.Validation.Attributes;
+
+/// <summary>
+/// Sets maximum required property value
+/// </summary>
+[AttributeUsage(AttributeTargets.Property)]
+public class MaxAttribute : ValidationAttribute
+{
+	/// <summary>
+	/// Initializes a new instance of the <see cref="MaxAttribute"/> class.
+	/// </summary>
+	/// <param name="maxValue">Maximum value of the property.</param>
+	/// <param name="errorMessage">The error message.</param>
+	/// <param name="isMessageFromStringTable">if set to <c>true</c> [is message from string table].</param>
+	public MaxAttribute(IComparable maxValue, string? errorMessage = null, bool isMessageFromStringTable = true) : base(errorMessage, isMessageFromStringTable) => MaxValue = maxValue;
+
+	/// <summary>
+	/// Gets or sets the maximum value of the property.
+	/// </summary>
+	/// <value>
+	/// The maximum value of the property.
+	/// </value>
+	public IComparable MaxValue { get; }
+
+	/// <summary>
+	/// Validates the specified property value.
+	/// </summary>
+	/// <param name="value">The object value.</param>
+	/// <param name="propertyInfo">Information about the property containing this attribute.</param>
+	/// <param name="resolver">The objects resolver, useful if you need to retrieve some dependencies to perform validation.</param>
+	public override void Validate(object? value, PropertyInfo propertyInfo, IDIResolver resolver)
+	{
+		if (value is not IComparable comparableValue || comparableValue.GetType() != MaxValue.GetType())
```

Comment:
**comparableValue.GetType() != MinValue.GetType()**
Maybe we should remove this check? this may cause implicit behavior

Reply:
That needed to exclude situation where we will compare not matching types. For example `int` with `long`.